src/btwa_api/main.py

```diff
@@ -48,6 +54,12 @@
     return response
 
 
+@app.middleware("http")
+async def auth_handler(request: Request, call_next, 
+                       credentials: HTTPBasicCredentials = Depends(security), db: Database = Depends(get_db)):
+    check_user_authenticated(db, credentials)
```

Comment:
Given this is an `async` fn, shouldn't the `check_user_authenticated` (which contains blocking IO code) call here be offloaded to a thread pool? What do you think @Kobzol ?

Reply:
Good point. It should be enough to make the middleware function sync.

Reply:
So for sync functions, FastAPI will run them on a threadpool by itself, right?

Reply:
Yes.